src/Component/Card/Card.tsx

```diff
@@ -0,0 +1,50 @@
+import "./styles.scss";
+import filledStar from "../../assets/icons/filled_star.svg";
+import emptyStar from "../../assets/icons/empty_star.svg";
+import { CardInterface } from "../../resources/interfaces";
+import nisIcon from "../../assets/icons/nis.svg";
+
+export const Card = ({
+  pic,
+  title,
+  desc,
+  rate,
+  price,
+  type,
+}: CardInterface) => {
+  const stars = [];
+
+  if (rate) {
+    for (let i = 0; i < 5; i++) {
+      if (i < rate) {
+        stars.push(<img key={i} src={filledStar} alt="Filled star" />);
+      } else {
+        stars.push(<img key={i} src={emptyStar} alt="Empty star" />);
+      }
+    }
+  }
+
+  return (
+    <div className="card_container">
+      <img src={pic} className="pic" />
+      <section className="text_container">
+        <h1>{title}</h1>
+        {type && <img src={type} className="type_icon" />}
+        <section className="desc_container">
+          <h3>{desc}</h3>
+          <div className="rating">{stars}</div>
+          {price && (
+            <div className="price_container">
+              <span className="line"></span>
+              <span className="price">
+                <img src={nisIcon} alt="NIS Icon" className="nis-icon" />{" "}
+                {price}
+              </span>
+              <span className="line"></span>
+            </div>
+          )}
+        </section>
+      </section>
+    </div>
+  );
+};
```

Comment:
Think of a way to make it more flexible. Remember the lesson on scaling horizontally – what if tomorrow we need to support a new type of card? With your current setup, you'd have to manually add new props and more conditions every time.

Instead, you could allow common props like the image, title, and bottom section to be passed in as child component. This way, whoever is rendering the card can design and add whatever they want (as the bottom section), or choose not to include certain elements at all.